compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BIRPackageSymbolEnter.java

```diff
@@ -1619,11 +1619,12 @@ public BType readType(int cpI) throws IOException {
                         ignoreAttachedFunc();
                     }
                     int funcCount = inputStream.readInt();
-                    for (int i = 0; i < funcCount; i++) {
-                        //populate intersection type object functions
-                        if (isImmutable(objectSymbol.flags) && Symbols.isFlagOn(flags, Flags.ANONYMOUS)) {
+                    if (isImmutable(objectSymbol.flags)) {
+                        for (int i = 0; i < funcCount; i++) {
```

Comment:
Redundant loop here. Assuming the change was to avoid multiple flag checks; move the flag check to a var instead and use it in the if/else